### PR TITLE
neo v0.8.0 compatible

### PR DESCRIPTION
**Pull request by dizcza**
Fixes https://github.com/NeuralEnsemble/elephant/issues/253

### CODE REVIEW

File: elephant/neo_tools.py [hunk 1]
```diff
@@ -45,17 +45,14 @@ def extract_neo_attrs(obj, parents=True, child_first=True,
...
-                # These would be duplicate
-                if a not in [_[0] for _ in obj._necessary_attrs + obj._recommended_attrs]:
-                    if "array_annotations" not in attrs:
-                        attrs["array_annotations"] = {}
-                    attrs["array_annotations"][a] = obj.array_annotations[a].copy()
-        except AttributeError:
-            pass
+    if not skip_array and hasattr(obj, "array_annotations"):
+        # Exclude labels and durations (and maybe other attributes) that are
+        # handled as array_annotations. These would be duplicate.
```
Comment (jpgill86) [inline elephant/neo_tools.py]: [According to](https://github.com/NeuralEnsemble/elephant/issues/253#issuecomment-539903686) @apdavison, epoch labels and durations were not assimilated into array_annotations with Neo 0.8.0, so this commend should be updated.

Comment (dizcza) [reply elephant/neo_tools.py]: Updated the comment

**Comment by jpgill86** [comment]
> Need to wait till [NeuralEnsemble/python-neo#746](https://github.com/NeuralEnsemble/python-neo/issues/746) is resolved and released with a new version.

I'm not quite sure how the NeuralEnsemble/python-neo#746 impinges on elephant.

In that issue, you mention functions called "`cut_segment_by_epoch`" and "`seg_time_slice`". Are these elephant functions that are intended to be dropped in the next elephant release because their job is now handled in Neo 0.8.0, but because the performance is unsatisfactory you don't want to drop elephant's implementations yet? I'm having trouble finding them in elephant.

Sorry I'm not understanding!

**Comment by dizcza** [comment]
Some Elephant users have developed and since then have been using these functions that now became a part of neo. But since I know them, I can mention this behavior and caution all other users in the following Elephant release.
As you probably saw, it's not only about `cut_segment_by_epoch` and `seg_time_slice`. And this is only one example I ran, I'm not sure how other examples are affected. But true, it should not harden the development and it's better to make minor releases often than waiting for a change.

Let's merge it.

**Comment by dizcza** [comment]
Added todos for the fix of https://github.com/NeuralEnsemble/python-neo/issues/749

**Pull request merged**